docs/guides/gitlab.md

```diff
@@ -0,0 +1,268 @@
+
+## Configure a Pomerium Route
+
+Edit `config.yaml` and add a route for GitLab:
+
+```yaml
+  - from: https://gitlab.localhost.pomerium.io
+    to: http://gitlab
+    preserve_host_header: true
+    policy:
+      - allow:
+          or:
+            - domain:
+                is: example.com
+```
+
+Once the route is applied, you should be able to access GitLab from `https://gitlab.localhost.pomerium.io`. Note that when using Docker, you may need to restart the Pomerium container to apply the changes as file change detection can be finicky on mounted docker volumes.
+
+Use `grep` within the container to find the default root password:
+
+```bash
+sudo docker exec -it gitlab grep 'Password:' /etc/gitlab/initial_root_password
+```
+
+### Configure TCP Connections
+
+1. An additional route will provide an encrypted TCP tunnel through which users can securely access code using Git:
+
+  ```yaml
+    - from: tcp+https://gitlab.localhost.pomerium.io
+      to: tcp://gitlab:22
+      policy:
+        - allow:
+            or:
+              - groups:
+                  has: devs@example.com
+  ```
+
+1. Once this route is applied, users can create an encrypted connection using [pomerium-cli] or the [Pomerium Desktop] app:
+
+    ::::: tabs
+    :::: tab Pomerium-CLI
+    ```bash
+    pomerium-cli tcp gitlab.pomerium.localhost.io:22 --listen :2202
+    ```
+
+    The `--listen` key defines what port the tunnel listens at locally.
+    ::::
+    :::: tab Pomerium Desktop
+    ![An encrypted tunnel for git connections using Pomerium Desktop](./img/gitlab/gitlab-desktop-connection.png)
+    ::::
+    :::::
+
+1. Add the tunneled connection as a remote:
+
+    ```bash
+    git remote add gitlab-tunneled ssh://git@127.0.0.1:2202/username/project-name
+    ```
+
+Now when you first initiate a `pull`, `push`, or `fetch` command your web browser will open to authenticate and authorize the connection.
+
+## Identity Management
+
+While GitLab has a [JWT OmniAuth provider], it only accepts a JSON web token (**JWT**) as a callback url parameter and not as an HTTP header as [provided by pomerium](/reference/readme.md#pass-identity-headers). If your IdP is a [supported Omniauth provider](https://docs.gitlab.com/ee/integration/omniauth.html#supported-providers), you can integrate it directly to GitLab to re-use your current session, but it will require signing in twice, once with Pomerium and again with GitLab:
+
+<video controls  muted="true" playsinline="" width="100%" control=""><source src="./img/gitlab/gitlab-login.webm" type="video/webm">
+Your browser does not support the video tag.
+</video>
+
+## Upstream TLS
+
+As part of a complete [zero-trust][Background] security model, all connections should be encrypted and [mutually authenticated][mTLS]. An important step in this process is configuring GitLab to serve content to Pomerium using HTTPS.
```

Comment:
```suggestion
As part of a complete [zero trust][Background] security model, all connections should be encrypted and [mutually authenticated][mTLS]. An important step in this process is configuring GitLab to serve content to Pomerium using HTTPS.
```
We are standardizing with "zero trust" (no hyphen, 2 words) because it's the most common variant for SEO purposes.